### PR TITLE
adding fluent vapor code

### PR DESCRIPTION
**Pull request by loganwright**

### CODE REVIEW

File: Sources/Droplet+Preparations.swift [hunk 1]
```diff
@@ -0,0 +1,28 @@
+
+import Fluent
+import Vapor
+
+extension Droplet {
+    /// The preparations that should be run by the droplet
+    public var preparations: [Preparation.Type] {
+        get {
+            guard let existing = storage["fluentvapor-preparations"] as? [Preparation.Type] else { return [] }
```
Comment (tanner0101) [inline Sources/Droplet+Preparations.swift]:
```swift
let preparationsKey = "fluentvapor-preparations"
```

File: Sources/Model.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
+//import JSON
+//import Fluent
+//import TypeSafeRouting
+//import HTTP
+//
+//public protocol Model: Entity, JSONRepresentable, StringInitializable, ResponseRepresentable { }
```
Comment (tanner0101) [inline Sources/Model.swift]: Still need to move `StringInitializable` to core

File: Sources/Provider.swift [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    public init(config: Settings.Config) {}
+
+    public func beforeRun(_ drop: Droplet) {
+        let prepare = Prepare(console: drop.console, preparations: drop.preparations, database: drop.database)
+        drop.commands.insert(prepare, at: 0)
+
+        // TODO: Propagate error up when Providers have `beforeRun` throwing
+        /// Preparations run everytime to ensure database is configured properly
+        do {
+            try prepare.run(arguments: drop.arguments)
```
Comment (tanner0101) [inline Sources/Provider.swift]: What happens if you do `vapor run prepare` ? does it run it twice?

**Review by tanner0101 — APPROVED** [review]
I would just run the prepare code directly in `beforeRun` but this looks like it can work.

File: Sources/Provider.swift (hunk 3 above)
Comment (loganwright) [reply Sources/Provider.swift]: Great point, need to handle that case.

**Pull request merged**